lib/bridge/IrcHandler.js

```diff
@@ -201,6 +207,55 @@ IrcHandler.prototype.onMessage = Promise.coroutine(function*(req, server, fromUs
         return;
     }
 
+    // For topics, set matrix topic as AS bot user
+    if (action.type === "topic") {
```

Comment:
This PR makes it so that there is no "common code" shared between topic changes and incoming messages anymore. Please can you split out everything in this `if` block into its own function like `onTopic` and call it from the `IrcEventBroker` to make this clearer.